### PR TITLE
Added interface to store chat conversations between AI and Human.

### PR DESCRIPTION
**Pull request by pontiyaraja**
- Added an interface to store the conversations history in mongodb. 
- And added Test case. 
- Configures pprof in texa app, and created make file to manage the build. 
- By pprof saved the running time of Texa APP, found there is too many `fmt.Println` 
- commented those lines. 
 
- Before Removing print statement
 Screenshot from 2019-04-10 12-32-03
 
- After Removing print statement
Screenshot from 2019-04-10 12-35-56

### CODE REVIEW

**Comment by pontiyaraja** [comment]
@0zAND1z

**Comment by 0zAND1z** [comment]
> @0zAND1z

Hi @pontiyaraja , thanks for the PR!

Can we have the `travis.yml` file added to this PR?
Here is an example that can help: https://github.com/0zAND1z/go-tara/blob/ab462ec70b77688d761c0928e5b7f770bc8650a8/.travis.yml

FYI, you can make good use of the benchmark & unit tests by executing them in Travis CI/CD.
Let me know if you need help here.

File: storage/chatStore.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+//[{userINput:"", AIOutput:""}, ....]
+type ChatItem struct {
+	UserInput string `bson:"userInput"`
+	AIOutPut  string `bson:"AIOutput"`
+}
+
+//formJson takes input as string array and forms the chatItem array
+func formJson(chathistory []string) []ChatItem {
```
[the remark this answers is not in the document]
Comment (0zAND1z) [reply storage/chatStore.go]: Cc @pontiyaraja. Please make the above recommended change.

File: storage/chatStore.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		session, err = mgo.Dial("localhost")
+		if err != nil {
+			return session, err
+		}
+	}
+	c.Unlock()
+	return session.Copy(), err
+}
+
+//[{userINput:"", AIOutput:""}, ....]
```
[the remark this answers is not in the document]
Comment (0zAND1z) [reply storage/chatStore.go]: Cc @pontiyaraja. Please make the above recommended change.

File: storage/chatStore.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/*
```
[the remark this answers is not in the document]
Comment (0zAND1z) [reply storage/chatStore.go]: Cc @pontiyaraja. Please make the above recommended change.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# The name of the executable (default is current directory name)
+TARGET := $(shell echo $${PWD})
+install:
+	go get
+	go install
+run:
+	go run main.go
+fmt:
+	gofmt -s -l $(TARGET)
+test:
```
Comment (0zAND1z) [inline Makefile:12]: You can actually run this `make test` in the proposed travis config file.

File: README.md [hunk 5]
```diff
@@ -14,33 +14,156 @@ The system uses a number of open source projects to work properly:
...
-TEXA requires [Go Lang](https://golang.org/)  v1.7+ to run.
+### Go:
 
-Install the dependencies, devDependencies and start the server.
+TEXA is written in [Go](http://golang.org) v1.7+. If you don't have a Go
+development environment, please [set one up](http://golang.org/doc/code.html).
+
+### Redis: 
+
+Texa using the [texalib]
```
Comment (0zAND1z) [inline README.md]: How about we reform the sentence as:

 "TEXA uses Redis to store all the mathematical data used by the `texalib` package, depended by the TEXA API. You can install [redis](https://redis.io/) and learn more about redis [here](https://redis.io/topics/quickstart#redis-quick-start)." ?

File: README.md [hunk 6]
```diff
@@ -14,33 +14,156 @@ The system uses a number of open source projects to work properly:
...
+to store the slabs, CatPages and MtsPages, in [redis](https://redis.io/)
+[Redis quick start](https://redis.io/topics/quickstart#redis-quick-start).
+
+### MongoDB:
+
+TEXA using [MongoDB](https://docs.mongodb.com/manual/installation) to persist the conversation between
```
Comment (0zAND1z) [inline README.md]: Again, suggesting to reform the sentence as:

"TEXA persists all your chat history in a local MongoDB instance, keeping your interrogation data private. You can install [MongoDB](https://docs.mongodb.com/manual/installation) and learn more about MongoDB [here](https://docs.mongodb.com/manual/tutorial/getting-started/)."

File: README.md [hunk 7]
```diff
@@ -14,33 +14,156 @@ The system uses a number of open source projects to work properly:
...
 
-``$ go run main.go``
+Run make file, by the following order
+
+```sh
+   make install
+   make test
+   make run
+
+   make fmt
```
Comment (0zAND1z) [inline README.md]: Can we have this command & the following commands below abstracted into a separate section called CONTRIBUTE ?

File: storage/bench0 [hunk 8]
```diff
@@ -0,0 +1,2 @@
+PASS
```
Comment (0zAND1z) [inline storage/bench0]: Is this file needed in the source?

**Comment by 0zAND1z** [comment]
Also, @pontiyaraja , I see a few binaries in the commits such as `storage/cpu.out`, `storage/profile001.pdf`, `storage/storage.test` and `storage/profile.out`.

Can we have them removed, if not required in the source repo? Thanks.

**Review by 0zAND1z — CHANGES_REQUESTED** [review]

**Comment by pontiyaraja** [comment]
> > @0zAND1z
> 
> Hi @pontiyaraja , thanks for the PR!
> 
> Can we have the `travis.yml` file added to this PR?
> Here is an example that can help: https://github.com/0zAND1z/go-tara/blob/ab462ec70b77688d761c0928e5b7f770bc8650a8/.travis.yml
> 
> FYI, you can make good use of the benchmark & unit tests by executing them in Travis CI/CD.
> Let me know if you need help here.

Okay, I will add it

File: storage/bench0 (hunk 8 above)
Comment (pontiyaraja) [reply storage/bench0]: no we can remove it

File: README.md [hunk 9]
```diff
@@ -3,7 +3,7 @@
 TEXA is a novel framework backed by a mathematical oriented theory(Texa Theory, refer to /texa-docs) for interacting, evaluating and assessing AIs.
```
Comment (0zAND1z) [inline README.md:7]: You may also add the Travis CI Badge, once you have the .travis successfully setup.
Badge code: [Build Status](https://travis-ci.org/TexaProject/texa)

**Review by 0zAND1z — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (pontiyaraja) [reply README.md]: ok

**Review by 0zAND1z — APPROVED** [review]

**Comment by 0zAND1z** [comment]
- All things passing
